object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java

```diff
@@ -150,6 +156,47 @@ boolean returnsThis(final MethodInvocationTree tree) {
     return rrType != null && rrType.hasAnnotation(This.class);
   }
 
+  /**
+   * Given a tree, returns the method that the tree should be considered as calling. Returns
+   * "withOwners" if the call is {@code withFilters(..., new Filter("owner"), ...)}. Otherwise,
+   * returns its first argument.
+   *
+   * <p>Package-private to permit calls from {@link ObjectConstructionTransfer}.
+   *
+   * @return either the first argument, or "withOwners" if the call is {@code withFilters(..., new
+   *     Filter("owner"), ...)}
+   */
+  String adjustMethodNameUsingValueChecker(
+      final String methodName, final MethodInvocationTree tree) {
+    if (useValueChecker) {
+      if ("withFilters".equals(methodName)) {
+        for (Tree filterTree : tree.getArguments()) {
+          // find a constructor invocation, digging through receivers
+          while (filterTree != null && filterTree.getKind() == Tree.Kind.METHOD_INVOCATION) {
+            filterTree =
+                TreeUtils.getReceiverTree(((MethodInvocationTree) filterTree).getMethodSelect());
+          }
+          if (filterTree == null) {
+            continue;
+          }
+          if (filterTree.getKind() == Tree.Kind.NEW_CLASS) {
+            ExpressionTree argTree = ((NewClassTree) filterTree).getArguments().get(0);
```

Comment:
I would find the name `withFiltersArg` (the argument to `withFilters`) clearer than `argTree`.
But, this and the following 10 lines feel like they should be abstracted out:  "the value is exactly 'owner'".  If there is not a method that does that, I suggest adding it.

Reply:
Such a method would belong in the Value Checker, but I don't think it exists.

I could make a pull request to the Value Checker, but I think that would delay this PR unnecessarily. It's also not clear it would be useful elsewhere.

Reply:
Please make that independently, and add a brief comment here to use that after a release is made.

Reply:
https://github.com/typetools/checker-framework/pull/2726